ml_peg/analysis/bulk_crystal/phonons/analyse_phonons.py

```diff
@@ -0,0 +1,647 @@
+    "avg_freq": "ω_avg",
+    "min_freq": "ω_min",
+    "S": "S",
+    "F": "F",
+    "C_V": "C_V",
+}
+BZ_COLUMN = "Avg BZ MAE"
+STABILITY_COLUMN = "Stability F1"
+STABILITY_THRESHOLD = -0.05 * THZ_TO_K
+T_300K_INDEX = 3  # Index for 300K in thermal properties (0, 75, 150, 300, 600)
+
+
+def _load_band_structure(file_path: Path) -> dict[str, Any] | None:
+    """
+    Load a serialized band-structure dataset.
+
+    Parameters
+    ----------
+    file_path
+        Absolute path to the ``npz`` file containing the band structure.
+
+    Returns
+    -------
+    dict[str, Any] | None
+        Parsed dictionary describing distances/frequencies, or ``None`` if the
+        file cannot be read.
+    """
+    try:
+        with open(file_path, "rb") as f:
+            return pickle.load(f)
+    except OSError as exc:
+        print(f"Failed to load band structure from {file_path}: {exc}")
+        return None
+
+
+def _load_dos(file_path: Path) -> tuple[np.ndarray, np.ndarray] | None:
+    """
+    Load frequencies and densities of states.
+
+    Parameters
+    ----------
+    file_path
+        Location of the ``npz`` file storing DOS data.
+
+    Returns
+    -------
+    tuple[np.ndarray, np.ndarray] | None
+        Pair of ``(frequency_points, total_dos)`` arrays, or ``None`` if the
+        file content could not be opened.
+    """
+    try:
+        with open(file_path, "rb") as f:
+            dos_dict = pickle.load(f)
+        return dos_dict["frequency_points"], dos_dict["total_dos"]
+    except OSError as exc:
+        print(f"Failed to load DOS from {file_path}: {exc}")
+        return None
+
+
+def _load_thermal_properties(file_path: Path) -> dict[str, Any] | None:
+    """
+    Load vibrational thermodynamic properties.
+
+    Parameters
+    ----------
+    file_path
+        Path to the JSON file containing entropy, free energy, and heat
+        capacity arrays.
+
+    Returns
+    -------
+    dict[str, Any] | None
+        Mapping of thermal quantities keyed by temperature, or ``None`` if the
+        JSON file is missing.
+    """
+    try:
+        with open(file_path, encoding="utf8") as f:
+            return json.load(f)
+    except OSError as exc:
+        print(f"Failed to load thermal properties from {file_path}: {exc}")
+        return None
+
+
+def _get_mp_ids() -> list[str]:
+    """
+    Return all Materials Project IDs with reference data.
+
+    Returns
+    -------
+    list[str]
+        Sorted collection of MP identifiers discovered in the reference DFT
+        directory.
+    """
+    ref_dir = CALC_PATH / "DFT"
+    if not ref_dir.exists():
+        print(f"ERROR: Reference directory not found: {ref_dir}")
+        print(f"Expected location: {ref_dir.absolute()}")
+        return []
+
+    mp_ids = set()
+    for file_path in ref_dir.glob("mp-*_thermal_properties.json"):
+        mp_id = file_path.stem.replace("_thermal_properties", "")
+        mp_ids.add(mp_id)
+
+    return sorted(mp_ids)
```

Comment:
Just to note, we do this elsewhere too, but since these are strings, it will sort things differently to how you'd expect, e.g. `['1', '10', '11', '8', '9']`. Is this a problem?